libravatar/libravatar.php

```diff
@@ -81,26 +70,15 @@ function libravatar_addon_admin(&$a, &$o)
 		'monsterid' => DI::l10n()->t('monster face'),
 		'wavatar' => DI::l10n()->t('computer generated face'),
 		'retro' => DI::l10n()->t('retro arcade style face'),
+		'robohash' => DI::l10n()->t('roboter face'),
+		'pagan' => DI::l10n()->t('retro adventure game character'),
 	];
 
-	// Show warning if PHP version is too old
-	if (! version_compare(PHP_VERSION, '5.3.0', '>=')) {
-		$o = '<h5>' .DI::l10n()->t('Warning') .'</h5><p>';
-		$o .= DI::l10n()->t('Your PHP version %s is lower than the required PHP >= 5.3.', PHP_VERSION);
-		$o .= '<br>' .DI::l10n()->t('This addon is not functional on your server.') .'<p><br>';
-		return;
-	}
-
-	// Libravatar falls back to gravatar, so show warning about gravatar addon if enabled
-	$r = q("SELECT * FROM `addon` WHERE `name` = '%s' and `installed` = 1",
-		DBA::escape('gravatar')
-	);
-	if (count($r)) {
+	if (Addon::isEnabled('gravatar')) {
 		$o = '<h5>' .DI::l10n()->t('Information') .'</h5><p>' .DI::l10n()->t('Gravatar addon is installed. Please disable the Gravatar addon.<br>The Libravatar addon will fall back to Gravatar if nothing was found at Libravatar.') .'</p><br><br>';
 	}
 
 	// output Libravatar settings
-	$o .= '<input type="hidden" name="form_security_token" value="' . BaseModule::getFormSecurityToken("libravatarsave") .'">';
```

Comment:
What didn't work with this?

Reply:
It always said that the security token had been invalid. And since only this and the `gravatar` addon are using this stuff (and also because there seens to be some other protection in the core) I removed it.